### PR TITLE
Fix inconsistent snapshots on different platforms

### LINKED ISSUE #1786: Running snapshot tests with -u on different platforms modifies paths
## Description
When a project has snapshots created on Linux, all the paths in the markdown files (and presumably somewhere in the `.snap` file) contain forward slashes. When you rerun the exact same tests on Windows with `-u`, e.g. to update a single changed snapshot, every `.md` and `.snap` file in the project gets modified so the paths contain backslashes instead. This makes it difficult to review commit diffs, because it looks at first glance like every snapshot file in the entire project has somehow been affected by the change.

Running tests with `-u` shouldn't touch the separators in the file paths at all, and in fact ideally the paths should be normalized (e.g. to always use forward slashes regardless of platform).

### Test Source
N/A. You can run the following on Windows to reproduce the issue:

```
git clone https://github.com/masaeedu/fp
cd fp
git checkout 33ef555
yarn
yarn test -u
git status
```

Every markdown and snap file is modified, whereas nothing should be modified.

### Error Message & Stack Trace
N/A

### Config
Not relevant to the issue, although you can take a look at the config in the repo linked above.

### Command-Line Arguments

```
ava -u
```

## Relevant Links

- https://github.com/masaeedu/fp

## Environment

Tell us which operating system you are using, as well as which versions of Node.js, npm, and AVA. Run the following to get it quickly:

```
node -e "var os=require('os');console.log('Node.js ' + process.version + '\n' + os.platform() + ' ' + os.release())"
Node.js v10.0.0
win32 10.0.15063

ava --version
0.25.0

npm --version
1.6.0
```

### PR DESCRIPTION
**Pull request by sharkykh**
Fixes #1786.
Made the changes as per @novemberborn's suggestions: https://github.com/avajs/ava/issues/1786#issuecomment-387119194 and https://github.com/avajs/ava/issues/1786#issuecomment-390485436.

### CODE REVIEW

File: lib/snapshot-manager.js [hunk 1]
```diff
@@ -128,7 +129,7 @@ function generateReport(relFile, snapFile, entries) {
 	const buffers = combined.buffers;
 	let byteLength = combined.byteLength;
 
-	const header = Buffer.from(`# Snapshot report for \`${relFile}\`
+	const header = Buffer.from(`# Snapshot report for \`${slash(relFile)}\`
 
 The actual snapshot is saved in \`${snapFile}\`.
```
[the remark this answers is not in the document]
Comment (sharkykh) [reply lib/snapshot-manager.js]: From what I can tell, `snapFile` only ever holds the snap file**name``.
https://github.com/avajs/ava/blob/f7fce6a81cf8ee77d0e4823e4ff8961d5097da17/lib/snapshot-manager.js#L392

**Comment by sharkykh** [comment]
Thanks for the review!
I will look into adding tests.

File: tap-snapshots/test-integration-snapshots.js-TAP.test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/* IMPORTANT
+ * This snapshot file is auto-generated, but designed for humans.
+ * It should be checked into source control and tracked carefully.
+ * Re-generate by setting TAP_SNAPSHOT=1 and running tests.
+ * Make sure to inspect the output below.  Do not ignore changes!
+ */
+'use strict'
+exports[`test\\integration\\snapshots.js TAP snapshots are indentical on different platforms > report file contents matches snapshot 1`] = `
```
Comment (sharkykh) [inline tap-snapshots/test-integration-snapshots.js-TAP.test.js]: This is also problematic: It's using Windows paths since I'm on Windows.

**Comment by novemberborn** [comment]
I wouldn't use TAP snapshots for this. Instead commit the expected files in one location, and then use AVA to generate new files in a different location. Then, compare them to make sure they're byte-for-byte the same.

**Comment by sharkykh** [comment]
Closing and re-opening to restart the failing Travis build.

**Comment by novemberborn** [comment]
Thanks @sharkykh, will try and have a look soon.

**Comment by sharkykh** [comment]
> Could you elaborate?

```
C:\Github\ava>npm run lint

> ava@1.0.0-beta.7 lint C:\Github\ava
> xo


  lib\ava-files.js:161:4
  ‼  161:4   Unexpected todo comment.              no-warning-comments

  lib\beautify-stack.js:29:1
  ‼   29:1   Missing JSDoc for parameter stack.    valid-jsdoc
  ‼   29:1   Missing JSDoc @returns for function.  valid-jsdoc
  ‼   66:2   Unexpected todo comment.              no-warning-comments

  lib\babel-pipeline.js:127:2
  ‼  127:2   Unexpected todo comment.              no-warning-comments
  ‼  153:3   Unexpected todo comment.              no-warning-comments
  ‼  163:61  Unexpected todo comment.              no-warning-comments
  ‼  166:62  Unexpected todo comment.              no-warning-comments
  ‼  169:66  Unexpected todo comment.              no-warning-comments

  lib\runner.js:188:4
  ‼  188:4   Unexpected todo comment.              no-warning-comments

  test\api.js:951:19
  ‼  951:19  Unexpected fixme comment.             no-warning-comments

  test\ava-files.js:77:2
  ‼   77:2   Unexpected todo comment.              no-warning-comments

  test\fixture\symlinkfile.js:1:1
  ×    1:1   Parsing error: Unexpected token .

  12 warnings
  1 error
```

I enabled `core.symlinks` in `.gitconfig`, but now I get errors when using git, so I don't know :confused: 
```
/c/Github/ava (master)
$ git reset --hard master
error: unable to create symlink test/fixture/symlink: Permission denied
error: unable to create symlink test/fixture/symlinkfile.js: Permission denied
fatal: Could not reset index file to revision 'master'.
```

**Comment by novemberborn** [comment]
Maybe it'll work better in a fresh clone?

**Comment by sharkykh** [comment]
Nope, it's just Windows being Windows, apparently.
https://github.com/git-for-windows/git/wiki/Symbolic-Links
Running `git reset --hard master` as an administrator works (created the symbolic links).

**Pull request merged**